File: manage_bash_aliases.mjs
import fs from "fs";
import path from "path";
import readline from "readline";

// Define the path to the aliases file
const aliasesFilePath = path.join(process.cwd(), "aliases.txt");

fs.readFile(aliasesFilePath, "utf8", (err, bashAliasesContent) => {
  if (err) {
    console.error(`Error reading aliases file: ${err.message}`);
    process.exit(1);
  }

  const filePath = path.join(
    process.env.HOME || process.env.USERPROFILE,
    ".bash_aliases"
  );

  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  fs.access(filePath, fs.constants.F_OK, (err) => {
    if (err) {
      // File does not exist, create it
      fs.writeFile(filePath, bashAliasesContent, (err) => {
        if (err) throw err;
        console.log("The ~/.bash_aliases file has been created.");
      });
    } else {
      // File exists, ask if it should be overwritten
      rl.question(
        "The ~/.bash_aliases file already exists. Do you want to overwrite it? (y (yes) / n (no)): ",
        (answer) => {
          if (
            answer.trim().toLowerCase() === "y" ||
            answer.trim().toLowerCase() === "yes"
          ) {
            fs.writeFile(filePath, bashAliasesContent, (err) => {
              if (err) throw err;
              console.log("The ~/.bash_aliases file has been overwritten.");
            });
          } else {
            console.log("The ~/.bash_aliases file was not overwritten.");
          }
          rl.close();
        }
      );
    }
  });
});

File: manage_bash_aliases.py
#!/usr/bin/env python3

from pathlib import Path

# Read the aliases from the text file
aliases_file_path = Path("aliases.txt")
if aliases_file_path.exists():
    bash_aliases_content = aliases_file_path.read_text()
else:
    raise FileNotFoundError(f"{aliases_file_path} not found.")

# Define the path to the .bash_aliases file
file_path = Path.home() / ".bash_aliases"

# Check if the .bash_aliases file exists and handle accordingly
if file_path.exists():
    overwrite = (
        input(
            "The ~/.bash_aliases file already exists. Do you want to overwrite it? (y (yes) / n (no)): "
        )
        .strip()
        .lower()
    )
    if overwrite == "y" or overwrite == "yes":
        file_path.write_text(bash_aliases_content)
        print("The ~/.bash_aliases file has been overwritten.")
    else:
        print("The ~/.bash_aliases file was not overwritten.")
else:
    file_path.write_text(bash_aliases_content)
    print("The ~/.bash_aliases file has been created.")
